### ISSUE TITLE
Add a text gradient block support

### ISSUE THREAD

**Issue by carolinan** [WordPress/gutenberg#30982]
## What problem does this address?

Paragraph blocks do not have a gradient color option.
This feature was requested during user testing.

EDIT: This was done for backgrounds in https://github.com/WordPress/gutenberg/pull/43119 but there is also a suggestion that we should be able to add graidents for text!

## What is your proposed solution?

Enable support for gradient controls for the paragraph block.

### Disclaimer
I am not personally invested in this 🖌️ 🌈 🙈 🎨 , but I have been asked to open issues that came up during a contributor day.

**Comment by mtias** [comment]
Was this request for background gradient or for gradient mask on the text itself?

**Comment by carolinan** [comment]
It was not specified so I assumed they meant background. Asking for clarification.

But gradient _headings_ could be nice... 
https://codepen.io/leocampos/pen/pPwwNR
https://codepen.io/saclark/pen/EepFx

**Comment by mtias** [comment]
Yes! Those would be neat to support

**Comment by luminuu** [comment]
I'd love to see the ability to put gradients on paragraphs and headlines too! Maybe a background gradient would be nice too but I'd prefer text gradient over background gradient.

**Comment by scruffian** [comment]
Reopening so we don't forget about the text aspect of this.

**Comment by phil-cariad** [comment]
I would love to see some cool text gradients make their way into blocks. A powerful header gradient would be very nice!

**Comment by phil-sola** [comment]
I would love to see some cool text gradients make their way into blocks. A powerful header/paragraph gradient would be very nice!

**Comment by jnz31** [comment]
+1

**Comment by CreativeDive** [comment]
Hey, this could be solved by using a block style variation:

```php
register_block_style( 'core/heading',
	array(
		'name'  => 'background-clip-text',
		'label' => __( 'Text Background', 'panther' ),
	)
);
```

```css
*:is(h1, h2, h3, h4, h5, h6).is-style-background-clip-text:not(.has-background) {
	background: linear-gradient(#000, #CCC);
}

*:is(h1, h2, h3, h4, h5, h6).is-style-background-clip-text {
	padding: 0!important;
	color: transparent!important;
	background-clip: text!important;
	-webkit-background-clip: text!important;
	-webkit-text-fill-color: transparent!important;
}
```

**Comment by andrewserong** [comment]
Just a tiny comment to note that a dependency for this feature idea is resolving the following:

* https://github.com/WordPress/gutenberg/issues/32787

**Comment by mtias** [comment]
@andrewserong I'm not sure this should block adding text gradients, it would just mean for the time being you cannot use a separate background color on the same element that you apply the text mask, and you'd need to use a group wrapper or something.

**Comment by andrewserong** [comment]
Thanks Matías — if we lift that as a requirement I think that unblocks digging into a WIP to explore this feature, good idea 👍

**Comment by andrewserong** [comment]
Just capturing a thought from this morning so I don't forget: something to try for this one could be to dynically inject a `span` around the text content when gradient text is enabled, and apply the rules to that `span`, as that could allow block-level backgrounds and text gradient to co-exist.

**Comment by mtias** [comment]
It'd be nice to not affect the markup, though.

**Comment by andrewserong** [comment]
Gotcha, in that case making gradient text and backgrounds mutually exclusive at the block level could be a good first step (that could then be enhanced further down the track once the background stuff is untangled)

**Comment by mtias** [comment]
Yes, I think that'd be a solid path to get it started.

**Comment by aaronrobertshaw** [comment]
Picked this back up this week.

There are two PRs up for review now rebased on the latest trunk.

- [#77141](https://github.com/WordPress/gutenberg/pull/77141) registers the `background.backgroundClip` support, no UI.
- [#77142](https://github.com/WordPress/gutenberg/pull/77142) adds the controls on top. Screencasts and screenshots are on that one.

@jasmussen the gradient sits below Color in the Typography panel with its own flyout, as you suggested. It is an optional item rather than shown by default, which is the one place it differs from your sketch. Worth a look on the PR if you think it should be visible from the start.

@jameskoster text color is now disabled with a tooltip while a gradient is applied, and the clip control is a select with a preview per option rather than the tile grid you found hard to read.

Heading and Paragraph opt in. The clip control itself stays hidden unless a theme sets `settings.background.backgroundClip`, so the text gradient ships without exposing the rougher box clipping UI.

[#76171](https://github.com/WordPress/gutenberg/pull/76171) is closed, superseded by the two above.
